topic: "joy"
base:
  submodes:
    steering:
      linear_x: {axis: 1, scale: 0.2}
      angular_z: {axis: 0, scale: 0.4}
      throttle: {axis: 3, scale: 2.0}
      enable: 
        - 0 #Deadman's switch = fire buttons
    strafing:
      linear_x: {axis: 1, scale: 0.2}
      linear_y: {axis: 0, scale: 0.1} #depends on button press_to_strafe
      throttle: {axis: 3, scale: 2.0}
      enable: 
        - 1 #striped button
        - 0 #Deadman's switch = fire buttons
neck:
  pan_simple: {axis: 4, scale: 0.15}
  tilt_simple: {axis: 5, scale: 0.15}
  submodes:
    control:
      pan: {axis: 4, scale: 3.0}
      tilt: {axis: 5, scale: -3.0}
      enable: 
        - 0 #Deadman's switch = fire buttons
  predefined_pose_stepper: 4 #Press this button to go to a next predefined pose
  predefined_poses:
    base:     {pan: 0,    tilt: -1.2, button: 9}
    driving:  {pan: 0,    tilt: -0.5, button: 8}
    left:     {pan: 0.5,  tilt: -1.2, button: 6}
    right:    {pan: -0.5, tilt: -1.2, button: 7}
arms:
  submodes:
    linear:
      linear_x: {axis: 0, scale: 0.6}
      linear_y: {axis: 1, scale: -0.6}
      linear_z: {axis: 2, scale: 0.6}
      enable: 
        - 0 #Deadman's switch = fire buttons
    angular:
      angular_x: {axis: 1, scale: 1.2}
      angular_y: {axis: 2, scale: 1.2}
      angular_z: {axis: 0, scale: -1.2}
      enable: 
        - 1 #striped button 
        - 0 #Deadman's switch = fire buttons
  open_close: 5 #Button 6 on the base of the joystick
next_mode: 2
previous_mode: 3
deadzone: 0.2 #axis values smaller than this value will be set to 0.0. For a spacenav, its hard to control only 1 axis
